src/HotChocolate/Core/src/Types/Types/Relay/Attributes/IDAttribute.cs

```diff
@@ -171,7 +171,9 @@ public class IDAttribute<T> : DescriptorAttribute
     /// <inheritdoc cref="IDAttribute{T}"/>
     public IDAttribute()
     {
-        TypeName = typeof(T).Name;
+        var type = typeof(T);
+
+        TypeName = type.GetCustomAttribute<GraphQLNameAttribute>()?.Name ?? type.Name;
```

Comment:
We have to go deeper than this as it does not take into account things like fluent types and so on. We need to hook into the config system and after types are named resolve this type.

Reply:
I will prototype something.